addons/website_slides/models/slide_channel.py

```diff
@@ -229,10 +280,19 @@ def _get_default_enroll_msg(self):
     partner_ids = fields.Many2many(
```

Comment:
Ok, it makes sense, but if we want the invited members to access the course, we would have to update the course ACL instead to something like 'partner_all_ids', '=', user.partner_id.id, am I right?

Reply:
Actually, this seems much more difficult that what I thought, I am almost stuck here. How to put a domain, since it is on res.partner ? Also, on res_partner, there is slide_channel_ids, that also must be filtered to only use enrolled partner_ids. The problem is that they directly use the relational table slide_channel_partner, so we would have to change them in computes fields ? It seems overkill... Any idea?

Reply:
It makes senes, indeed. This has been an interesting point to work on. There are two things to notice/do/discuss here :


A. ACLS -- Quick reminder : the idea is to allow invited members to see the course, even if it is members-only (original goal of this task). Also, invited members can see the preview slides and browse the course before enrolling.

A1 If we if we want the invited members to access the course, we would have to update the rule to something like 'partner_all_ids', '=', user.partner_id.id, am I right? I've proposed a solution in the code.
A2 The slide ACL would also be modified to allow invited members to access preview slides before enrolling. Also changed.
A3 resource ACL stays the same

B. PARTNER_IDS stays the same -- the partners of the course who are enrolled.
There is now a distinction with partner_all_ids, also containing invited partners.
The relational table between courses and partners contains all attendees.

B1 Before, partner_ids was a M2M field using the relational table between courses and partners. However, this is not valid anymore since it also contains invited partners.
-> Therefore, I had to change to a computed_field with a search method instead.
B2 partner_all_ids use the relational table as partner_ids did, without trouble.
B3 In res_partner.py, slide_channel_ids used the same relational table. The change has also been done there. Should we consider only the courses the partner has been enrolled to, knowing that the field is used in mass_mailing to contact attendee? I guessed we should only consider enrolled one when contacting attendees. Does that make sense?

NB: we cannot just put a domain on the res.partner since the member_status is on the slide.channel.partner, not on the res_partner.